x-pack/plugins/security_solution/public/timelines/components/timeline/helpers.tsx

```diff
@@ -253,3 +261,94 @@ export const focusUtilityBarAction = (containerElement: HTMLElement | null) => {
 export const resetKeyboardFocus = () => {
   document.querySelector<HTMLAnchorElement>('header.headerGlobalNav a.euiHeaderLogo')?.focus();
 };
+
+interface OperatorHandler {
+  field: string;
+  isExcluded: string;
+  value: string | number | Array<string | number>;
+}
+
+export const handleISOperator = ({
+  browserFields,
+  field,
+  isExcluded,
+  isFieldTypeNested,
+  type,
+  value,
+}: OperatorHandler & {
+  browserFields: BrowserFields;
+  isFieldTypeNested: boolean;
+  type?: DataProviderType;
+}) => {
+  if (!isStringOrNumberArray(value)) {
+    return `${isExcluded}${
+      type !== DataProviderType.template
+        ? buildISQueryMatch({ browserFields, field, isFieldTypeNested, value })
+        : buildEXISTSQueryMatch({ browserFields, field, isFieldTypeNested })
+    }`;
+  } else {
+    return `${isExcluded}${field} : ${JSON.stringify(value)}`;
+  }
+};
+
+export const handleISONEOFOperator = ({ field, isExcluded, value }: OperatorHandler) => {
+  if (isStringOrNumberArray(value)) {
+    return `${isExcluded}${buildISONEOFQueryMatch({ field, value })}`;
+  } else {
+    return `${isExcluded}${field} : ${JSON.stringify(value)}`;
+  }
+};
+
+export const buildISQueryMatch = ({
+  browserFields,
+  field,
+  isFieldTypeNested,
+  value,
+}: {
+  browserFields: BrowserFields;
+  field: string;
+  isFieldTypeNested: boolean;
+  value: string | number;
+}): string => {
+  if (isFieldTypeNested) {
+    return convertNestedFieldToQuery(field, value, browserFields);
+  } else if (checkIfFieldTypeIsDate(field, browserFields)) {
+    return convertDateFieldToQuery(field, value);
+  } else {
+    return `${field} : ${isNumber(value) ? value : escapeQueryValue(value)}`;
+  }
+};
+
+export const buildEXISTSQueryMatch = ({
+  browserFields,
+  field,
+  isFieldTypeNested,
+}: {
+  browserFields: BrowserFields;
+  field: string;
+  isFieldTypeNested: boolean;
+}): string => {
+  return isFieldTypeNested
+    ? convertNestedFieldToExistQuery(field, browserFields).trim()
+    : `${field} ${EXISTS_OPERATOR}`.trim();
+};
+
+export const buildISONEOFQueryMatch = ({
```

Comment:
consider using the following casing for this function name: `buildIsOneOfQueryMatch`

Reply:
changed